lib/hawkular/base_client.rb

```diff
@@ -175,6 +178,16 @@ def tenant_header
       headers
     end
 
+    # @!visibility private
+    def connect_error(fault)
+      if fault.is_a?(SocketError)
+        HawkularConnectionException.new(fault.to_s)
+      elsif [Errno::ECONNREFUSED, Errno::ECONNRESET, Errno::EHOSTUNREACH,
+             Errno::ENETDOWN, Errno::ENETUNREACH, Errno::ETIMEDOUT].include?(fault.class)
+        HawkularConnectionException.new(fault.to_s, fault.class::Errno)
```

Comment:
I think we are safe for all that are POSIX compliant ( http://man7.org/linux/man-pages/man3/errno.3.html ). Windows since NT is Posix as well as real Unixoids (like Linux).